### DISCUSSION TITLE
How to define a color scheme for each element added in the seed_elements.

### DISCUSSION THREAD

**Discussion by GuiCruz** [OpenDrift/opendrift#1442] [Q&A] [answered]
I have two groups of elements, and I would like to know if it's possible to define a color for each group, such as EAST or WEST. When plotting the graph with o.plot(), one group would be blue and the other red.

#east
`
o.seed_elements(lon = -52.068594, lat = -32.164484, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.067167, lat = -32.168863, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.065834, lat = -32.173146, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.064216, lat = -32.177524, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.062979, lat = -32.181902, number=30, radius =100, z= 0, time=reader.start_time)
`
#west
`
o.seed_elements(lon = -52.098480, lat = -32.167340, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.096291, lat = -32.171337, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.094102, lat = -32.175335, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.091723, lat = -32.179332, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.089534, lat = -32.183425, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.087249, lat = -32.187232, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.085060, lat = -32.191229, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.082586, lat = -32.195322, number=30, radius =100, z= 0, time=reader.start_time)
o.seed_elements(lon = -52.080492, lat = -32.199224, number=30, radius =100, z= 0, time=reader.start_time)
`

**Comment by knutfrode** [comment, marked as answer]
You can use element property `origin_marker` for this purpose:
```
from datetime import datetime
from opendrift.models.oceandrift import OceanDrift

lon_e  =[-52.068594, -52.067167, -52.065834, -52.064216, -52.062979]
lat_e = [-32.164484, -32.168863, -32.173146, -32.177524, -32.181902]

lon_w = [-52.098480, -52.096291, -52.094102, -52.091723, -52.089534, -52.087249, -52.085060, -52.082586, -52.080492]
lat_w = [-32.167340, -32.171337, -32.175335, -32.179332, -32.183425, -32.187232, -32.191229, -32.195322, -32.199224]

time = datetime.now()
o = OceanDrift()
num_per_location = 30
o.seed_elements(lon=lon_e*num_per_location, lat=lat_e*num_per_location, 
                time=time, radius=100, origin_marker=0)
o.seed_elements(lon=lon_w*num_per_location, lat=lat_w*num_per_location,
                time=time, radius=100, origin_marker=1)
o.run(steps=1)
o.animation(color='origin_marker', fast=True, buffer=.08)
```

Note that plotting does not support coloring elements with property, so I used here animation method for that purpose, and had to trick by simulate one timestep.

![seed](https://github.com/user-attachments/assets/f40822fc-eb97-4fe1-b46d-e0362dbfac7c)

**Reply by GuiCruz** [reply]
Thank you!

**Comment by GuiCruz** [comment]
Unfortunately, I need to plot the graphs with the trajectories, and `o.animation` won’t be helpful.

I saw that it’s possible to save the files using the function: `o.write_netcdf()`.

I think I'll create a NetCDF file with the results for each trajectory and use matplotlib to plot the trajectories in the way I want. Thank you very much for the quick response, Knutfrode.

I have another question: is it possible to add shapefiles when using `o.plot()`?

**Comment by knutfrode** [comment]
There is no `write_netcdf()` method, but output will be saved to netCDF-file if you provide `filename` to run method.

After making a simulation, you can add `linecolor="origin_marker"` to `o.plot(   )` to color lines as shown on animation above.

There is no method to add shapefiles with built in plot method, but you can create more customized figures with the TrajAn package under development:
https://opendrift.github.io/trajan/
